### PR TITLE
OCPBUGS-2346: refactor(dvo_metrics): remove name and namespace from dvo metrics

### PR DESCRIPTION
**Pull request by rluders**
This PR implements the filtering for DVO Metrics information. The idea is that the DVO metrics will have the `name` and `namespace` information on the cluster unless the IO archive will not have this information by default.

**How to test it?**

- Create a Cluster (clusterbot or AWS)
- Install DVO from its repository (or you can run it locally)
    - Allow openshift-monitoring to access the metrics:
```sh
$ oc process --local NAMESPACE='openshift-monitoring' -f deploy/openshift/network-policies.yaml | oc create -f -
```
[deploy/openshift/network-policies.yaml](https://github.com/app-sre/deployment-validation-operator/blob/master/deploy/openshift/network-policies.yaml)

- Forward metrics service:
```
$ kubectl port-forward service/deployment-validation-operator-metrics 8383:8383 -n openshift-operators
```
- Get this PR and you can run it locally
- Create a standard httpserver deployment at the `default` namespace, and scale it down to 2 instances

**You should expect to:**
- The DVO metrics being generated here
```sh
$ curl http://localhost:8383/metrics
```
- IO should gather this info at the `dvo_metrics` and remove the namespace and name from the metric

## Categories

- [ ] Bugfix
- [X] Enhancement
- [ ] Backporting
- [ ] Others (CI, Infrastructure, Documentation)

## Sample Archive
N/A

## Documentation
N/A

## Unit Tests
WIP

## Privacy

Yes. There are no sensitive data in the newly collected information.

## Breaking Changes

Yes. It removes the `name` and `namespace` from DVO metrics.

## References

https://issues.redhat.com/browse/CCXDEV-9328
https://issues.redhat.com/browse/OCPBUGS-2346

### CODE REVIEW

File: pkg/gatherers/clusterconfig/dvo_metrics.go [hunk 1]
```diff
@@ -129,10 +130,21 @@ func gatherDVOMetricsFromEndpoint(
 		}
 	}()
 
-	prefixedLines, err := utils.ReadAllLinesWithPrefix(dataReader, dvoMetricsPrefix)
+	prefixedLines, err := utils.ReadAllLinesWithPrefix(dataReader, dvoMetricsPrefix, func(b []byte) []byte {
+		filterProps := []string{"name", "namespace"}
+
+		for _, f := range filterProps {
+			re := regexp.MustCompile(fmt.Sprintf(`(?m)(,?%s=[^\,\}]*")`, f))
```
[the remark this answers is not in the document]
Comment (rluders) [reply pkg/gatherers/clusterconfig/dvo_metrics.go:137]: Good point.

File: pkg/utils/read_lines_with_prefix_test.go [hunk 2]
```diff
@@ -14,7 +14,7 @@ func Test_ReadAllLinesWithPrefix(t *testing.T) {
 		"prefix_fourth_line",
 		"prefix_last_line",
 	}, string(MetricsLineSep)))
-	lines, err := ReadAllLinesWithPrefix(reader, []byte("prefix_"))
+	lines, err := ReadAllLinesWithPrefix(reader, []byte("prefix_"), nil)
```
Comment (tremes) [inline pkg/utils/read_lines_with_prefix_test.go:17]: It would be good to introduce some new test using the filter function.

**Comment by tremes** [comment]
OK. I was finally able to verify this locally. There's no test but we should try to update the logic to use the Prometheus exposition format in future.
/lgtm

**Pull request merged**